src/pkg/project/dao/dao.go

```diff
@@ -39,6 +39,8 @@ type DAO interface {
 	GetByName(ctx context.Context, name string) (*models.Project, error)
 	// List list projects
 	List(ctx context.Context, query *q.Query) ([]*models.Project, error)
+	// Lists the roles of user for the specific project
+	ListRoles(ctx context.Context, projectID int64, userID int, groupIDs ...int) ([]int, error)
```

Comment:
It's not very clear the relationship between uid and gid?  it seems the uid MUST be a member of each GID provided?  If that's the case, would it be better if we combine the data in one data object. 

I recall models.User has the values but a lot of other irrelevant attributes.

Reply:
It's not very well to import `pkg/user/models` here. 

We are in agreement that one package in `pkg` doesn't import other packages in the `pkg` directory.